en/docs/references/ballerina-expression-syntax.md

```diff
@@ -1,422 +1,485 @@
+        lname: "Stallone",
+        family: [
+            {fname: "James", lname: "Stallone"},
+            {fname: "Blake", lname: "Stallone"},
+            {fname: "Kyle", lname: "Stallone"},
+            {fname: "Taylor", lname: "Stallone"}
+        ]
+    }
+```
+
+To get the length of a JSON array, use the following:
+
+```ballerina
+personRecordArray.length(); 
+```
+
+### JSON access
+
+You can use the field access(.) and optional field access (?.) to access the `json` type in Ballerina. In this case, the type checking happens at runtime. The return type of this operation is a union of `json` and `error`. 
+    
+!!! info
+    The examples in this section will assume the following json is already defined.
+    
+    ```ballerina
+         json person = {
+             fname: "Alex",
+             lname: "Jane",
+             address: {
+                 line: "20 Palm Grove",
+                 city: "Colombo 03",
+                 country: "Sri Lanka"
+             }
+         }
+    ```    
+
+- Field access(.) 
+
+    You can use the field access operator to access record fields of the `json` type. If the key is invalid or the field access occurs on `null`, it returns `error`.
+   
+    ###### Examples:
+
+    To access a field, use the following:    
+
+    ```ballerina
+       check person.fname;
+    ```
+    To access a field in a nested level, use the following:
+ 
+     ```ballerina
+        check person.address.city;
+     ```
 
-Query parameters are used in an HTTP request to send additional inputs to consider when processing the request. You can use the `request` variable's `getQueryParamValue` function to read the passed query parameter.
+- Optional field access (?.) 
 
-#### Examples
+    You can use the optional field access operator to access record fields, including any optional fields. If the key is invalid or the field access occurs on `null`, it returns `null ` or `()`.
 
-- Reading the query parameter value given the query parameter name (`category`).
+    ```ballerina
+    check person?.address?.city;
+    ```
 
-  ```ballerina
-  request.getQueryParamValue("category")
-  ```
-  
-- Reading the query parameter value by passing a variable that contains the query parameter name. Note the omission of the double-quotes.
+### JSON/Record/Map conversion
 
-  ```ballerina
-  request.getQueryParamValue(queryParamName)
-  ```
+Ballerina records, maps, and JSON objects hold records. Records are collections of fields, and you can access each field value by a key. You can write expressions converting from one type to another.
 
-### Reading the cookies available in the request
+###### Examples:
 
-HTTP cookies keep stateful information against a client. You can use the `request` variable's `getCookies` function to read cookies present in the request as an array.
+To convert the record `movieTheRevenant` to `json`, use the following:
 
-#### Example
+```ballerina
+      check movieTheRevenant.cloneWithType(json);
```

Comment:
I think we need to consider explaining `toJson` and `fromJsonWithType` too (or instead).

https://docs.central.ballerina.io/ballerina/lang.value/1.0.0/functions#toJson
https://docs.central.ballerina.io/ballerina/lang.value/1.0.0/functions#fromJsonWithType

Reply:
This is not yet supported by the Console. Hence took down the content.